src/controllers/BookingController.ts

```diff
@@ -56,35 +48,38 @@ export class BookingController implements IBookingController {
       },
       capabilities
     );
-    const bookingAvailability = {
-      id: availabilityModel.id,
-      localDateTimeStart: availabilityModel.localDateTimeStart,
-      localDateTimeEnd: availabilityModel.localDateTimeEnd,
-      allDay: availabilityModel.allDay,
-      openingHours: availabilityModel.openingHours,
-    };
-
-    const optionId = schema.optionId;
-    const optionModel = productWithAvailabilityModel.findOptionModelByOptionId(optionId);
-
-    if (optionModel === null) {
-      throw new InvalidOptionIdError(optionId);
-    }
-
-    schema.unitItems.forEach((bookingUnitItemSchema: BookingUnitItemSchema) => {
-      const unitId = bookingUnitItemSchema.unitId;
-      const unitModel = optionModel.findUnitModelByUnitId(unitId);
+    const bookingModel = await this.bookingService.createBookingModel(
+      schema,
+      productWithAvailabilityModel,
+      availabilityModel
+    );
+    this.checkRestrictions(bookingModel.optionModel, schema.unitItems);
 
-      if (unitModel === null) {
-        throw new InvalidUnitIdError(unitId);
+    if (schema.orderId !== undefined) {
```

Comment:
i think the logic is not 100% accurate, you don't check the capability there at all

there are these cases: 
1. cart capability not requested - ignore creating order
2. cart capability requested and orderId not provided - create order and attach the booking to it
3. cart capability and orderId provided - try to get order with that id and attach the booking to it, if not found, throw 400